doc/tutorials/understanding-error-messages.md

```diff
@@ -0,0 +1,161 @@
+# Understanding Error Messages
+
+Errors can happen when trying to mount with insufficient permissions, missing plugin configurations or when you want to write
+specifications for your configurations and test it out or many other occasions. The error message gives information on what the problem is and possibly how
+to resolve them.
+
+Here is an example how you can trigger an error for an invalid IP entry in the `hosts` file:
+
+```sh
+sudo kdb mount --with-recommends /etc/hosts system/tests/hosts hosts
+sudo kdb set system/tests/hosts/ipv4/my 88.198.134.1777
+# RET: 5
+# Sorry, module network issued the error C03200:
+# Validation Semantic: name: system/tests/hosts/ipv4/my value: 88.198.134.1777 message: Name or service not known
+
+#cleanup
+sudo kdb umount system/tests/hosts
+```
+
+In this example we mount the hosts file under `/etc/hosts` into Elektra. Elektra also validates
```

Comment:
Simply make assumptions (in the beginning of the tutorial) what you expect readers to know. It does not make sense to explain mounting here. The important part (why set fails), however, is not yet explained here.